rust/kernel/module_param.rs

```diff
@@ -71,11 +71,17 @@ pub trait ModuleParam: core::fmt::Display + core::marker::Sized {
         let arg = if val.is_null() {
             None
         } else {
+            // SAFETY: `val` is a valid CStr by the safety requirements of
+            //         this function
```

Comment:
Could you please follow a closer style to other files? e.g.

```suggestion
            // SAFETY: `val` is a valid `CStr` by the safety requirements of
            // this function.
```

Also, it is slightly confusing to say `val` is a valid `CStr`, since it is not a `CStr`.

More importantly, we have to show that `val` points to a NUL-terminated string (`val` could have been NULL according to the contract above, but we just checked for it). Also, we should explain why the contents are not mutated for the lifetime (this is called from the C side, so perhaps you need to expand the safety contract to say so).